### PR TITLE
Encapsulate user/sc address split in an enum

### PR DESCRIPTION
**Pull request by Ben-PH**
* [x] document all added functions
* [ ] try in sandbox /simulation/labnet
* [x] unit tests on the added/changed features
  * [x] make tests compile
  * [x] make tests pass 
* [x] add logs allowing easy debugging in case the changes caused problems
* [x] if the API has changed, update the API specification
* [ ] Update labnet to populate initial rolls and ledger with new address leader
* [ ] grep other parts of the codebase for wher "AU1" is used, and update.

### CODE REVIEW

**Comment by Ben-PH** [comment]
CI test fails because it relies on an update of the test repo. (changing the "A1..." string to "AU1...")

File: massa-pos-exports/src/pos_changes.rs
Comment (Ben-PH) [reply massa-pos-exports/src/pos_changes.rs:102]: Thanks for the catch.

This should use the serializer. If I was to make some assumptions, the use of extend was chosen due to type being a raw-wrapper around an array. This PR doesn't change that all that much, however that might not be the case for too long. With that change, the use of a serializer to handle this step will be important

**Comment by Ben-PH** [comment]
Unit tests pass locally. This PR depends on an update to `massa-unit-test-src`. The fix for local is to buuild that repo and put the result into the appropriate directory. As CI only builds from the repo, unedited, updating it is required to pass CI

File: massa-bootstrap/src/tests/tools.rs
Comment (AurelienFT) [inline massa-bootstrap/src/tests/tools.rs:243]: It would be better to have a comment on why we are setting the first byte to a magic value

File: massa-execution-worker/src/context.rs
Comment (AurelienFT) [inline massa-execution-worker/src/context.rs:414]: This should be an SC address here. I saw that in Address you used the same type for `UserAddress` and `SCAddress`. It's ok to define the structure SCAddress in the second time but we should open an issue to not forget before merging this but in this line we could use the variant SCAddress to avoid having a wrong line of code merged.

File: massa-pos-exports/src/pos_changes.rs
Comment (AurelienFT) [reply massa-pos-exports/src/pos_changes.rs]: It was just because we didn't took the time to make an `AddressSerializer` now that you made one it's nice to use it :)

**Comment by Leo-Besancon** [comment]
Sure, I'll look at the implementation! 👍

File: massa-execution-worker/src/context.rs
Comment (Ben-PH) [reply massa-execution-worker/src/context.rs]: I misunderstood your request. Now has `Address::SC(SCAddress(...))`

**Comment by Ben-PH** [comment]
I'll merge once @AurelienFT is happy with the remaining unresolved conversations.

**Review by AurelienFT — APPROVED** [review]
I resolved everything. Great work !!

**Pull request merged**
